src/server/pps/server/api_server.go

```diff
@@ -2564,23 +2564,35 @@ func (a *apiServer) deletePipeline(ctx context.Context, request *pps.DeletePipel
 			}); err != nil && !errutil.IsNotFoundError(err) {
 				return errors.Wrap(err, "error deleting pipeline repo")
 			}
-		} else if pipelineInfo.Details != nil {
-			// Remove branch provenance from output and meta
-			// we can only proceed if we have details
-			if _, err := a.env.PfsServer().CreateBranch(ctx, &pfs.CreateBranchRequest{
-				Branch: client.NewBranch(pipelineName, pipelineInfo.Details.OutputBranch),
-			}); err != nil && !errutil.IsNotFoundError(err) {
-				return err
+		} else {
+			// Remove branch provenance from output and then delete meta and spec repos
+			// this leaves the repo as a source repo, eliminating pipeline metadata
+			if err := a.txnEnv.WithWriteContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
```

Comment:
We should refactor this such that all of the Postgres state change when deleting a pipeline happens in one transaction.